src/explorer/AzureAccountTreeItem.ts

```diff
@@ -25,8 +26,14 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase {
     public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
         const ti: AzExtTreeItem | undefined = this.trialAppNode ?? await this.loadTrialAppNode();
         const children: AzExtTreeItem[] = await super.loadMoreChildrenImpl(clearCache, context);
+
         if (ti) {
             children.push(ti);
+        } else {
+            // Must check children[0] instanceof GenericTreeItem because if Azure account is still loading then isLoggedIn is false.
+            if (!this.isLoggedIn && children[0] instanceof GenericTreeItem) {
```

Comment:
I think you can move this logic into `loadTrialAppNode`  I think you could for this inside the  `(!loginSession)` condition.  I'm not sure how necessary the `children[0]` check is.  Did you run into where it would say "Loading..." and create new trial app?

Reply:
Ok after some investigating:
If I move the logic into `loadTrialAppNode` then I have to check `children[0]` (why I moved it out of `loadTrialAppNode` in the first place. **But** now I have found that if I leave the logic out of `loadTrialAppNode` then I don't have to check `children[0]`.

I moved it out since we already had access to `children` outside `loadTrialAppNode`.